### ISSUE TITLE
No way to generate \\{{var}}

### ISSUE THREAD

**Issue by fzerorubigd** [XaminProject/handlebars.php#113]
Correct result of (context is `['var' => 'value']`) : 

```
\\\{{var}}
````
is 
```
\value
```
but handlebars.php produce 
```
\\{{var}}
```

**Comment by fzerorubigd** [comment]
@JustBlackBird I can't get my head around this. I need to generate exactly one `\` before a variable. 
 handlebars.js use two `\\` to output a `\` but since its PHP, the the `\\` in any case compiled to `\` (before loading by engine, I can't understand why PHP translate `'\\'` to `\` even when I am in single qute!) 
So if i want exactly two, I should have `\\\` OR `\\\\` (WTF?) so this are wrong 
https://github.com/XaminProject/handlebars.php/blob/master/tests/Xamin/HandlebarsTest.php#L130
Any idea on this?

**Comment by JustBlackBird** [comment]
@fzerorubigd, according to [php manual](http://php.net/manual/en/language.types.string.php#language.types.string.syntax.single) `\\\{` inside of _single quotes_ is equal to `\\\\{`, thus [this](https://github.com/XaminProject/handlebars.php/blob/master/tests/Xamin/HandlebarsTest.php#L126) and [this](https://github.com/XaminProject/handlebars.php/blob/master/tests/Xamin/HandlebarsTest.php#L130) test cases do the same job.

To get a single backslash inside a PHP's string one should double it. So
```php
$s = "\\{{var}}";
```
puts `\{{var}}` string into `$s` variable and
```php
$s = "\\\\{{var}}";
```
puts `\\{var}`.

The problem is the current Handlebars.php parser does not replace `\\` in a template with `\`. You can check this behavior by using templates with `\Handlebars\Loader\FilesystemLoader`:
```handlebars
\{{var}}
```
will be rendered to `\{{var}}` while
```handlebars
\\{{var}}
```
becomes `\\value` but should be rendered to `\value`.

To summarize, you are right, there is no way to render `\\{{var}}` template correctly.

**Comment by majortom731** [comment]
A workaround would be to store a single backslash in a different variable and then use 

    {{backslash}}{{var}}

 inside the template.

**Comment by JustBlackBird** [comment]
@majortom731 it's a dirty hack. The problem should be fixed in the Core.

**Comment by fzerorubigd** [comment]
@majortom731 Actually, I use that in my template right now. but as @JustBlackBird said we need to fix it in core.

**Comment by JustBlackBird** [comment]
Here is a jsfiddle that illustrates Handlebars.js behavior: http://jsfiddle.net/cLxyge73/1/ . It seems that only slashes before opening curly brace (`{`) should be doubled.

**Comment by JustBlackBird** [comment]
@fzerorubigd could you please review #118 ?
